### PR TITLE
Mesh:  add constants for initial/default values of fields

### PR DESCRIPTION
**Pull request by MelVimL**
Changed default values to "0" instead of "-1" and moved them to private constant class variables.
I guess "-1" was used as an fail/error-state but i didn't seen any checks besides ">=0", "!=0" and "==0".

Also after reading the request which also fixes #334 I am not sure if this a useful solution. 

Unittests -> ok
Handpicked examples(Scenes, Animation) ->  #ok

### CODE REVIEW

**Comment by stephengold** [comment]
This isn't a good solution for #313, because the render statistics will still be incorrect.

If the statistics are -1, that indicates that the mesh hasn't been completely initialized. If such a mesh were passed to the renderer, it might be best to fail quickly.

**Comment by MelVimL** [comment]
Okay. I should read more thoroughly: **"If the stats value is -1 the mesh should be considered invalid and crash with exception."**

**Comment by stephengold** [comment]
We could either close this PR or else turn it into a refactoring PR that simply adds a bunch of constants. @MelVimL what's your preference?

If we pursue the latter course, I request that the constants not be public unless there's a good case for using them outside the package.

**Comment by MelVimL** [comment]
I like constants for defaults/inits, even the private ones.

File: jme3-core/src/main/java/com/jme3/scene/Mesh.java [hunk 1]
```diff
@@ -148,28 +149,44 @@ public boolean isListMode() {
             return listMode;
         }
     }
+    
+    /**
+     * Default Variables
+     */
+    private static final int DEFAULT_VERTEXT_ARRAY_ID = -1;
+    private static final CollisionData DEFAULT_COLLECTION_TREE = null;
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/scene/Mesh.java:157]: This should be `COLLISION` not `COLLECTION`.

File: jme3-core/src/main/java/com/jme3/scene/Mesh.java [hunk 2]
```diff
@@ -148,28 +149,44 @@ public boolean isListMode() {
             return listMode;
         }
     }
+    
+    /**
+     * Default Variables
+     */
+    private static final int DEFAULT_VERTEXT_ARRAY_ID = -1;
```
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/scene/Mesh.java:156]: This should be `VERTEX` not `VERTEXT`.

**Review by stephengold — CHANGES_REQUESTED** [review]
with apologies for the slow review

File: jme3-core/src/test/java/com/jme3/scene/mesh/MeshTest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.Test;
+
+import com.jme3.scene.Mesh;
+
+public class MeshTest {
+       @Test
+    public void testEmptyMesh() {
+        final Mesh mesh = new Mesh();
+
+        assertEquals(-1, mesh.getVertexCount());;
```
Comment (capdevon) [inline jme3-core/src/test/java/com/jme3/scene/mesh/MeshTest.java:14]: double semicolon ';;'  at the end of the instruction

Comment (MelVimL) [reply jme3-core/src/test/java/com/jme3/scene/mesh/MeshTest.java:14]: Changed it! Thank you for the Feedback!

File: jme3-core/src/main/java/com/jme3/scene/Mesh.java (hunk 2 above)
Comment (MelVimL) [reply jme3-core/src/main/java/com/jme3/scene/Mesh.java]: Fixed it! Sorry for the typos.

**Review by stephengold — APPROVED** [review]
Looks good.

**Comment by stephengold** [comment]
Unless there's further substantive discussion, I plan to integrate this in 24 hours.

**Comment by stephengold** [comment]
Oops! I just noticed the added file "MeshTest.java". Probably it should be deleted from this PR. If not deleted, it needs javadoc and a copy of the license.

@MelVimL would you please take care of it?

**Comment by MelVimL** [comment]
@stephengold  Added the requesting javadoc and license. I used the TestSpatial as a reference.

**Comment by stephengold** [comment]
Thank you @MelVimL . I plan to integrate this in a few hours.

**Comment by MelVimL** [comment]
@stephengold You're Welcome. Love the project!

**Pull request merged**